src/pages/Notifications/NotificationsLog/Page.tsx

```diff
@@ -0,0 +1,198 @@
+import { Card } from '@patternfly/react-core';
+import Main from '@redhat-cloud-services/frontend-components/Main';
+import {
+  Filter,
+  Operator,
+  toUtc,
+} from '@redhat-cloud-services/insights-common-typescript';
+import { format, sub, toDate } from 'date-fns';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { NotificationsLogDateFilterValue } from '../../../components/Notifications/NotificationsLog/NotificationsLogDateFilter';
+import NotificationsLogTable, {
+  DrawerType,
+  PaginationType,
+} from '../../../components/Notifications/NotificationsLog/NotificationsLogTable';
+import NotificationsLogToolbar from '../../../components/Notifications/NotificationsLog/NotificationsLogToolbar';
+import { PageHeader } from '../../../components/PageHeader';
+import { Messages } from '../../../properties/Messages';
+
+export type NotificationsPeriod = [Date | undefined, Date | undefined];
+
+const sortMapper = {
+  0: 'title',
+  2: 'time',
+};
+
+const RETENTION_DAYS = 14;
+
+const createFilter = (dateFilter, period) => {
+  let filterPeriod = [undefined, undefined] as [
+    Date | undefined,
+    Date | undefined
+  ];
+  const today = toUtc(new Date());
+  const DATE_FORMAT = 'yyyy-MM-dd';
+  const filter = new Filter();
+
+  let yesterday;
+  switch (dateFilter) {
```

Comment:
I am not a fan of `switch` statement, it might be good to move this out of the `createFilter` function and use object mapper
```
const filterPeriodMapper = (dateFilter) => {
  const today = toUtc(new Date());
  const yesterday = sub(toDate(today), {
        days: 1,
      })
  return {
    [NotificationsLogDateFilterValue.LAST_14]: [
         sub(toDate(today), {
            days: 14,
          }),
          today
      ],
     [NotificationsLogDateFilterValue.LAST_7]: [
       sub(toDate(today), {
          days: 7,
        }),
        today
     ],
    [NotificationsLogDateFilterValue.TODAY]: [today, today],
    [NotificationsLogDateFilterValue.YESTERDAY]: [yesterday, yesterday]
  }[dateFilter]
}
```

usaga can be as:
```
const filterPeriodMapper(dateFilter) || period;
```

Reply:
Fixed